internal/server/ironhawk/watch_manager.go

```diff
@@ -130,8 +131,9 @@ func (w *WatchManager) CleanupThreadWatchSubscriptions(t *IOThread) {
 }
 
 func (w *WatchManager) NotifyWatchers(c *cmd.Cmd, shardManager *shardmanager.ShardManager, t *IOThread) {
-	w.mu.Lock()
-	defer w.mu.Unlock()
+	// Use RLock instead as we are not really modifying any shared maps here.
+	w.mu.RLock()
+	defer w.mu.RUnlock()
```

Comment:
This change should significantly improve the through put of notifications on watches as RLocks can be acquired concurrently. The earlier code would only allow either one writer or reader to be processed at the same time.

With this change, concurrent users can read but only one writer to be processed.